.gitignore

```diff
@@ -5,3 +5,4 @@ node_modules
 *.tgz
 temp
 .idea
+index.json
```

Comment:
package-lock.json can be ignored.

Reply:
Could you explain why it can be ignored? I thought that this file should always be committed.

Reply:
Preventing conflict in Git and ...
> One key detail about package-lock.json is that it cannot be published, and it will be ignored if found in any place other than the toplevel package. It shares a format with npm-shrinkwrap.json, which is essentially the same file, but allows publication. This is not recommended unless deploying a CLI tool or otherwise using the publication process for producing production packages.

Source: https://docs.npmjs.com/files/package-lock.json

Reply:
Hmm, I always thought that this file is needed to preserve exact dependencies versions, of course there are sometimes conflicts with git, but event in this npm doc about `package-loc` there is this sentence `This file is intended to be committed into source repositories` , so what is the difference between committing to source repository and publishing it?

Reply:
I believe that the "publishing" is a reference to NPM.
@zeke What do you think about the package-lock.json ? (remove or not remove)

Reply:
package-lock.json should be kept in the repo, their goal is (most importantly) to make sure all collaborators / build system have the same dependency tree.